sensors/sensor.py

```diff
@@ -0,0 +1,67 @@
+import time
+
+try:
+    import pigpio
+except ImportError:
+    pass
+
+class Sensor:
+    MAX_RETRIES = 5
+
+    def __init__(self, bus, address, start_command):
+        self.__bus = bus
+        self.__address = address
+        self.__start_command = start_command
+
+        print("Attempting to connect to Raspberry Pi GPIO...")
```

Comment:
`logging` library suggests the use of `.log` for outputs like this and `print` for user interfaces.